Sprint 6/step 1

dmi-viktor opened the pull request:
Финальный 6-й спринт. 

Как всегда, рад Вашим подробным комментариям. Удачи)

Commit:
Добавил тестирование jest, cypress.
Удалил строки с console.log

Commit:
Добавил ссылку на опубликованный проект в readme

gennady-bars commented on README.md:
```diff
@@ -44,3 +44,5 @@ You don’t have to ever use `eject`. The curated feature set is suitable for sm
 
 To learn React, check out the [React documentation](https://reactjs.org/).
+
+Link to publication GitHub Pages https://dmi-viktor.github.io/react-burger/index.html
```
`Link to publication GitHub Pages` лучше поместить в самое начало `Readme`, чтобы сразу была видна эта ссылка

dmi-viktor replied:
Спасибо. Перенес в начало.

gennady-bars commented on src/services/reducers/burger-constructor.test.js:
```diff
@@ -0,0 +1,201 @@
+        expect(
+            burgerConstructorReducer(initialState, {
+                type: types.ADD_TO_CONSTRUCTOR,
+                ingredientData: {
```
Если в тестах повторяются одинаковые объекты с одинаковыми данными, то их нужно выносить в **константы**, чтобы потом использовать в коде тестов. Это упрощает быстрое изменение тестовых данных и избавляет от дублирования.


**Нужно исправить дублирование одинаковых данных во всех тестах**

dmi-viktor replied:
Спасибо. 
Надеюсь понял Ваше замечание до конца верно, просмотрел ВСЕ файлы, внес исправления, где посчитал нужным. Константы заведены в каждом файле, обобщать и выносить в общий файл не стал. Небольшие обьекты с 3-4мя полями выносить не стал, мне кажется что так удобнее.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная

* проект задеплоен
* все 36 юнит-тестов проходят успешно
* все cypress-тесты проходят успешно

 но есть некоторые недочеты: 

* при входе в задеплоенное приложение попадаю на страницу 404. Скрин http://joxi.ru/J2bzbWltqejZk2 . Нужно попробовать сделать так, чтобы перенаправлялся пользователь на главную, если не нашелся никакой совпадающий рут
* `Link to publication GitHub Pages` лучше поместить в самое начало `Readme`, чтобы сразу была видна эта ссылка
* Если в тестах повторяются одинаковые объекты с одинаковыми данными, то их нужно выносить в **константы**, чтобы потом использовать в коде тестов. Это упрощает быстрое изменение тестовых данных и избавляет от дублирования.
* Если какой-то элемент этого объекта меняется, то нужно использовать `spread`-операцию, чтобы  создать копию объекта, а нужное поле просто точечно изменять после этого 
* **Нужно исправить дублирование одинаковых данных во всех тестах**


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

dmi-viktor commented:
Что касается 404 при попытке перейти по адресу "https://dmi-viktor.github.io/react-burger/index.html". Замечание справедливое.  Если страницы не найдена, то по предыдущему ТЗ, должен выводить страницу 404. Чтобы выйти из данной ситуации, расширил число url для главной страницы:

`<Route path={["/", "/react-burger/index.html", "/react-burger"]} exact={true}>
     <HomePage />
</Route>`

Так же я понимаю что данное решение имеет место быть только для выбранного способа публикации, ввиду отсутствия собственного доменного имени и в дальнейшем подобные "костыли" подлежат чистке.

Commit:
Перенос строки в файле README.md
Вынес в константы обьемные повторяющиеся данные для каждого файла тестов.
Правки в маршрутизации для главной страницы.

gennady-bars commented:
> Что касается 404 при попытке перейти по адресу "https://dmi-viktor.github.io/react-burger/index.html". Замечание справедливое. Если страницы не найдена, то по предыдущему ТЗ, должен выводить страницу 404. Чтобы выйти из данной ситуации, расширил число url для главной страницы:
> 
> `<Route path={["/", "/react-burger/index.html", "/react-burger"]} exact={true}> <HomePage /> </Route>`
> 
> Так же я понимаю что данное решение имеет место быть только для выбранного способа публикации, ввиду отсутствия собственного доменного имени и в дальнейшем подобные "костыли" подлежат чистке.

Да) у GH Pages есть нюансы

gennady-bars commented on src/services/reducers/wsOrderHistoryReducer.test.js:
```diff
@@ -2,14 +2,39 @@ import { wsOrderHistoryReducer, initialState } from './wsOrderHistoryReducer.ts'
         expect(
             wsOrderHistoryReducer(initialState, {
-                type: 'WS_ORDER_HISTORY_CONNECTION_SUCCESS' // todo. Уважаемый ревьюер, использую строковое выражение, быть может я не понял и можно было сделать "интереснее"? 
+                type: 'WS_ORDER_HISTORY_CONNECTION_SUCCESS'
```
да, я сам проглядел, что в тестах сокетов используете строки, а не константы

Нужно использовать именно константные типы экшенов, чтобы их можно было легко изменить (переназвать и тд). И чтобы не ошибиться в буквах, а потом не гадать, чего тест не проходит

dmi-viktor replied:
Исправился.

gennady-bars reviewed (commented):
Уже почти все отлично, но:

Кое-что проглядел прошлый раз. Извиняюсь.

* Нужно использовать именно константные типы экшенов, чтобы их можно было легко изменить (переназвать и тд). И чтобы не ошибиться в буквах, а потом не гадать, чего тест не проходит

Commit:
константные типы экшенов для реализации websocket

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.


Удачи в Вашей новой профессии.

**Несколько советов для дальнейшего развития:**

* сверстайте несколько небольших проектов для себя, для закрепления знаний, и чтобы можно добавить их в портфолио
* ходите на собеседования: опыт прохождения собеседований не менее важен знаний теории, и Вы поймете, какие знания требуются на рынке, и где Вы проседаете
* читайте [habr](https://habr.com/ru/) и слушаете подкаксты: они очень хорошо расширяют кругозор и позволяют узнать, что происходит в отрасли:

https://soundcloud.com/everyonecan

https://soundcloud.com/podlodka

https://soundcloud.com/devschacht

https://soundcloud.com/begebot

https://soundcloud.com/web-standards

https://www.youtube.com/channel/UCZeU17nbVfzczAkJVTay9vw

dmi-viktor commented:
Спасибо)
